workflow/snakemake_rules/fetch_sequences.smk

```diff
@@ -19,8 +19,66 @@ Produces different final outputs for GISAID vs GenBank/RKI:
 """
 
 wildcard_constraints:
-    # Constrain GISAID pair names to YYYY-MM-DD-N
-    gisaid_pair = r'\d{4}-\d{2}-\d{2}(-\d+)?'
+    # Constrain GISAID pair names to "gisaid_cache" or YYYY-MM-DD-N
+    gisaid_pair = r'gisaid_cache|\d{4}-\d{2}-\d{2}(-\d+)?'
+
+
+if config.get("s3_src"):
+
+    rule fetch_gisaid_ndjson:
+        """
+        Fetch previously uploaded gisaid.ndjson if it exists.
+        This is a cache of the raw data from previous GISAID ingest(s).
+        If it doesn't exist, then just create an empty file.
+        """
+        output:
+            ndjson=temp("data/gisaid/gisaid_cache.ndjson"),
+        params:
+            s3_file=f"{config['s3_src']}/gisaid.ndjson.zst",
+        shell:
+            r"""
+            if $(./vendored/s3-object-exists {params.s3_file:q}); then
+                ./vendored/download-from-s3 {params.s3_file:q} {output.ndjson:q}
+            else
+                echo "{params.s3_file:q} does not exist, creating empty file."
+                touch {output.ndjson:q}
+            fi
+            """
```

Comment:
[question] 

Initially the cache will be the last full ndjson dump (`s3://nextstrain-ncov-private/gisaid.ndjson.zst`), then we'll collect this file and the (unprocessed) batch-downloaded-files (`concatenate_gisaid_ndjsons`) which produces a new (complete) ndjson. The code to upload of this to be the "new" cache isn't part of this PR - or did I miss it?

Do we need to store a copy of the last full ndjson flat-file in the processed (?) directory so we can re-ingest everything if we ever need to?

Reply:
Re:

> echo "{params.s3_file:q} does not exist, creating empty file."

Consider a hard error? Unless we plan on the path of allowing a previous flat-file in the unprocessed folder, the absence of the "cache" is not going to end well...